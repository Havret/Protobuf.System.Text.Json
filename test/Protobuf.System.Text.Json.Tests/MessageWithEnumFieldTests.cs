using System.Text.Json.Protobuf.Tests.Utils;
using Shouldly;
using SmartAnalyzers.ApprovalTestsExtensions;
using Xunit;

namespace System.Text.Json.Protobuf.Tests;

public class MessageWithEnumFieldTests
{
    [Fact]
    public void Should_serialize_message_with_enum_field()
    {
        // Arrange
        var msg = new MessageWithEnum
        {
            EnumField = TestEnum.FirstOption
        };
        var jsonSerializerOptions = TestHelper.CreateJsonSerializerOptions();
 
        // Act
        var serialized = JsonSerializer.Serialize(msg, jsonSerializerOptions);

        // Assert
        var approver = new ExplicitApprover();
        approver.VerifyJson(serialized);
    }

    [Fact]
    public void Should_deserialize_message_with_with_enum_field()
    {
        // Arrange
        var msg = new MessageWithEnum
        {
            EnumField = TestEnum.FirstOption
        };
        var jsonSerializerOptions = TestHelper.CreateJsonSerializerOptions();

        // Act
        var serialized = JsonSerializer.Serialize(msg, jsonSerializerOptions);
        var deserialized = JsonSerializer.Deserialize<MessageWithEnum>(serialized, jsonSerializerOptions);

        // Assert
        deserialized.ShouldNotBeNull();
        deserialized.ShouldBeEquivalentTo(msg);
    }
}