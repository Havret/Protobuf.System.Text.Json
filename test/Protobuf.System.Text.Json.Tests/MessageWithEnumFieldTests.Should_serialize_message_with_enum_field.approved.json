{
  "enumField": 1,
  "regularField": 0
}